### PR TITLE
[servicegraphprocessor/servicegraphconnector] Change metric names to match the spec

### PR DESCRIPTION
**Pull request by mapno**
**Description:**

Changes servicegraph metrics to match the spec

**Link to tracking Issue:**

Solves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18743 and https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16578

**Testing:**

Added tests to make sure that the metrics match the spec

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: processor/servicegraphprocessor/processor_test.go [hunk 1]
```diff
@@ -277,13 +277,18 @@ func verifyHappyCaseMetrics(t *testing.T, md pmetric.Metrics) {
 	assert.Equal(t, 1, sms.Len())
 
 	ms := sms.At(0).Metrics()
-	assert.Equal(t, 2, ms.Len())
+	assert.Equal(t, 3, ms.Len())
 
 	mCount := ms.At(0)
 	verifyCount(t, mCount)
 
-	mDuration := ms.At(1)
```
Comment (jpkrohling) [inline processor/servicegraphprocessor/processor_test.go:285]: Should this be kept for a couple of releases to allow people to upgrade to the new metric name? Which one would best represent the old one? Can you also add more info in the changelog telling folks what they should do not to have their dashboards broken?

Comment (mapno) [reply processor/servicegraphprocessor/processor_test.go:274]: Added the legacy metric names under a feature flag and improved the description. Thanks for the review.

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Sorry for taking so long to review this. It looks like there's a conflict now. Are you able to fix it? Ping me once you do, so that I can review it again. Thanks!

**Comment by mapno** [comment]
No worries! Rebased and ready for merge @jpkrohling

**Pull request merged**